### PR TITLE
Z `pow` + `inv`

### PR DESCRIPTION
**Pull request by jnsiemer**
This PR implements
- [x] pow
- [x] inv

including tests for `Z`.

### CODE REVIEW

**Review by Marvin-Beckmann — CHANGES_REQUESTED** [review]
Nice work, please consider changing the return value of inverse and the corresponding changes from my comment since the error would not be needed anymore

**Review by Marvin-Beckmann — APPROVED** [review]
just small changes you can probably directly take over

File: src/integer/z/properties.rs [hunk 1]
```diff
@@ -31,17 +35,46 @@ impl Z {
...
+        let mut out = Q::ZERO;
+        // the new fmpz value does not need to be cleared manually as it's small
+        // the fmpq instance does neither as the fmpq value is dropped automatically,
+        // but the numerator/ important part is kept alive
+        let self_fmpq = fmpq {
+            num: self.value,
+            den: fmpz(1),
+        };
+        unsafe { fmpq_inv(&mut out.value, &self_fmpq) };
+        Some(out)
```
[the remark this answers is not in the document]
Comment (jnsiemer) [reply src/integer/z/properties.rs:76]: I've just added this as it does remove the need to clone the fmpz value and is thus more efficient. The explanation/ comment was added as Sven told me to (intrusively) in the last PR.

[the remark this answers is not in the document]
Comment (Marvin-Beckmann) [reply src/integer/z/properties.rs:76]: okay, then its fine, but add this explanation, as. this makes it more clear

**Pull request merged**